lib/main.dart

```diff
@@ -357,14 +358,21 @@ class ProvenanceWalletApp extends StatefulWidget {
 }
 
 class _ProvenanceWalletAppState extends State<ProvenanceWalletApp> {
-  _ProvenanceWalletAppState() {
-    _setup();
-  }
+  bool _isSetupComplete = false;
 
   final _subscriptions = CompositeSubscription();
   final _navigatorKey = GlobalKey<NavigatorState>();
   var _authStatus = AuthStatus.noAccount;
 
+  @override
+  Future<void> didChangeDependencies() async {
```

Comment:
Is there a case in which we would want to call `_setup()` more than one time? I think it would actually throw an error when registering the same singleton twice. Probably `initState` should be used.

Reply:
`State.initState() must be a void method without an 'async' keyword.`

Which means nothing can be awaited in `initState`, so setup is running at the same time allowing a race condition to throw off the app if a dependency is requested early. That makes this code change pointless.

I've got an idea but it can't be done in `initState`.

Reply:
Could use a `Completer` and a `FutureBuilder`

Reply:
Would need to verify expected behavior of an error thrown in `_setup()`